app/assets/javascripts/discourse/app/controllers/topic.js

```diff
@@ -1302,32 +1306,45 @@ export default Controller.extend(bufferedProperty("model"), {
       return;
     }
 
-    if (this.retryRateLimited || times <= 0) {
+    if (this._retryRateLimited || times <= 0) {
       return;
     }
 
-    promise().catch((e) => {
-      const xhr = e.jqXHR;
-      if (
-        xhr &&
-        xhr.status === 429 &&
-        xhr.responseJSON &&
-        xhr.responseJSON.extras &&
-        xhr.responseJSON.extras.wait_seconds
-      ) {
-        let waitSeconds = xhr.responseJSON.extras.wait_seconds;
-        if (waitSeconds < 5) {
-          waitSeconds = 5;
-        }
+    if (this._retryInProgress) {
+      later(() => {
```

Comment:
I think @jjaffeux brought this up but we really need to store a handle to any timeouts (or `laters`) and cancel them accordingly.